### DISCUSSION TITLE
Use cases for snowflake_table resource

### DISCUSSION THREAD

**Discussion by ralfsantacruz** [Snowflake-Labs/terraform-provider-snowflake#707] [Q&A] [answered]
Hey all, I attended the CI/CD with Snowflake webinar yesterday and heard from Snowflake's principal architect that snowflake tables shouldn't really be managed by terraform. We're currently managing a handful of tables with terraform and I wanted to know other people's experience with this.

We find using snowflake_table resources to be really helpful, especially with the ability to add comments that live next to the table such that we can automatically generate data dictionaries. I can see this getting a bit unwieldy as we scale though.

**Comment by tampajohn** [comment, marked as answer]
In my experience it really comes down to the team, and their familiarity with infrastructure as code.  I've worked with data engineering teams that found managing all of their dbs/schemas/tables/rbac via TF to be amazing, but they were quite seasoned with TF and it was a part of their toolkit.  

I have also worked with clients that were much more comfortable with maintaining a plethora of SQL scripts, and happy letting Snowflake be the source of truth when it comes to schema... 

I would say that consistency is more important than whether you decide to use TF or not,  I would refrain from maintaining some tables in terraform, and some outside of terraform, unless you have a very clean boundary between them (like separate accounts, or RBAC that restricts TF from touching those tables that are outside of TF's control).
